tests/ImageSharp.Tests/Memory/Allocators/UniformUnmanagedMemoryPoolTests.cs

```diff
@@ -0,0 +1,379 @@
+                Assert.NotNull(handles);
+                handleUnits.Add(handles);
+                foreach (UnmanagedMemoryHandle array in handles)
+                {
+                    allHandles.Add(array);
+                }
+
+                // Allocate some memory, so potential new pool allocation wouldn't allocated the same memory:
+                cleanup.Register(Marshal.AllocHGlobal(128));
+            }
+
+            foreach (UnmanagedMemoryHandle[] arrayUnit in handleUnits)
+            {
+                if (arrayUnit.Length == 1)
+                {
+                    // Test single-array return:
+                    pool.Return(arrayUnit.Single());
+                }
+                else
+                {
+                    pool.Return(arrayUnit);
+                }
+            }
+
+            handles = pool.Rent(totalCount);
+
+            Assert.NotNull(handles);
+
+            foreach (UnmanagedMemoryHandle array in handles)
+            {
+                Assert.Contains(array, allHandles);
+            }
+
+            cleanup.Register(allHandles);
+        }
+
+        [Fact]
+        public void Rent_SingleBuffer_OverCapacity_ReturnsInvalidBuffer()
+        {
+            var pool = new UniformUnmanagedMemoryPool(7, 1000);
+            using var cleanup = new CleanupUtil(pool);
+            UnmanagedMemoryHandle[] initial = pool.Rent(1000);
+            Assert.NotNull(initial);
+            cleanup.Register(initial);
+            UnmanagedMemoryHandle b1 = pool.Rent();
+            Assert.True(b1.IsInvalid);
+        }
+
+        [Theory]
+        [InlineData(0, 6, 5)]
+        [InlineData(5, 1, 5)]
+        [InlineData(4, 7, 10)]
+        public void Rent_MultiBuffer_OverCapacity_ReturnsNull(int initialRent, int attempt, int capacity)
+        {
+            var pool = new UniformUnmanagedMemoryPool(128, capacity);
+            using var cleanup = new CleanupUtil(pool);
+            UnmanagedMemoryHandle[] initial = pool.Rent(initialRent);
+            Assert.NotNull(initial);
+            cleanup.Register(initial);
+            UnmanagedMemoryHandle[] b1 = pool.Rent(attempt);
+            Assert.Null(b1);
+        }
+
+        [Theory]
+        [InlineData(0, 5, 5)]
+        [InlineData(5, 1, 6)]
+        [InlineData(4, 7, 11)]
+        [InlineData(3, 3, 7)]
+        public void Rent_MultiBuff_BelowCapacity_Succeeds(int initialRent, int attempt, int capacity)
+        {
+            var pool = new UniformUnmanagedMemoryPool(128, capacity);
+            using var cleanup = new CleanupUtil(pool);
+            UnmanagedMemoryHandle[] b0 = pool.Rent(initialRent);
+            Assert.NotNull(b0);
+            cleanup.Register(b0);
+            UnmanagedMemoryHandle[] b1 = pool.Rent(attempt);
+            Assert.NotNull(b1);
+            cleanup.Register(b1);
+        }
+
+        public static readonly bool IsNotMacOS = !TestEnvironment.IsOSX;
+
+        // TODO: Investigate MacOS failures
```

Comment:
Are these intermittent? Restricted to certain target frameworks?

Reply:
This one was intermittent, the one above *seemed* to occur consistently on .NET 6, but nothing is proven without having a Mac ...
Here it's probably just a test issue because of the OS returning identical addresses in a follow-up allocation, despite my attempt to push them out by dummy allocations:

https://github.com/SixLabors/ImageSharp/blob/7ec7447808725f761dc7cba83fea676c4420158f/tests/ImageSharp.Tests/Memory/Allocators/UniformUnmanagedMemoryPoolTests.cs#L266-L267